plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/ServerPortExtension.java

```diff
@@ -0,0 +1,126 @@
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.wst.server.core.IServerType;
+import org.eclipse.wst.server.ui.wizard.ServerCreationWizardPageExtension;
+
+/**
+ * An extension that adds a server port field to the WTP server creation wizard page.
+ */
+public class ServerPortExtension extends ServerCreationWizardPageExtension {
+
+  private static final String APP_ENGINE_SERVER_TYPE_ID =
+      "com.google.cloud.tools.eclipse.appengine.standard.server"; //$NON-NLS-1$
+
+  @VisibleForTesting Label portLabel;
+  @VisibleForTesting Text portText;
+  @VisibleForTesting ControlDecoration portDecoration;
+
+  @VisibleForTesting Image informationImage;
+  @VisibleForTesting Image errorImage;
+
+  @Override
+  public void createControl(UI_POSITION position, Composite parent) {
+    if (position == UI_POSITION.BOTTOM) {
+      portLabel = new Label(parent, SWT.NONE);
+      portLabel.setVisible(false);
+      portLabel.setText(Messages.NEW_SERVER_DIALOG_PORT);
+
+      portText = new Text(parent, SWT.SINGLE | SWT.BORDER);
+      portText.setVisible(false);
+      portText.setText(String.valueOf(LocalAppEngineServerBehaviour.DEFAULT_SERVER_PORT));
+      portText.addVerifyListener(new PortChangeMonitor());
+      portText.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
+
+      portDecoration = new ControlDecoration(portText, SWT.LEFT | SWT.TOP);
+      portDecoration.hide();
+
+      FieldDecorationRegistry registry = FieldDecorationRegistry.getDefault();
+      informationImage =
+          registry.getFieldDecoration(FieldDecorationRegistry.DEC_INFORMATION).getImage();
+      errorImage = registry.getFieldDecoration(FieldDecorationRegistry.DEC_ERROR).getImage();
+    }
+  }
+
+  @Override
+  public void handlePropertyChanged(PropertyChangeEvent event) {
+    if (event != null && event.getNewValue() instanceof IServerType) {
+      IServerType serverType = (IServerType) event.getNewValue();
+      boolean showPort = APP_ENGINE_SERVER_TYPE_ID.equals(serverType.getId());
+      portLabel.setVisible(showPort);
+      portText.setVisible(showPort);
+      if (showPort) {
+        updatePortAndTriggerDecoration(portText.getText());
+      } else {
+        portDecoration.hide();
+      }
+    }
+  }
+
+  private class PortChangeMonitor implements VerifyListener {
+    @Override
+    public void verifyText(VerifyEvent event) {
+      String newText = portText.getText().substring(0, event.start)
+          + event.text + portText.getText().substring(event.end);
+      event.doit = updatePortAndTriggerDecoration(newText);
+    }
+  };
+
+  private boolean updatePortAndTriggerDecoration(String newPortString) {
+    if (newPortString.isEmpty()) {
+      showPortDecoration(informationImage, Messages.NEW_SERVER_DIALOG_EMPTY_PORT_FIELD);
+      serverWc.setAttribute(LocalAppEngineServerBehaviour.SERVER_ATTRIBUTE_PORT, 0);
```

Comment:
Sorry, I think I failed to communicate clearly. This variable is in the super class, which is Eclipse code. @briandealwis agreed that there is nothing we can do about it.